raster/src/main/scala/geotrellis/raster/GridExtent.scala

```diff
@@ -174,7 +178,7 @@ class GridExtent[@specialized(Int, Long) N: Integral](
     val colMax: N = Integral[N].fromLong {
       val colMaxDouble = mapXToGridDouble(subExtent.xmax)
 
-      if (math.abs(colMaxDouble - floorWithTolerance(colMaxDouble)) < GridExtent.epsilon)
+      if (math.abs(colMaxDouble - floorWithTolerance(colMaxDouble)) < Constants.DOUBLE_EPSILON)
```

Comment:
Actually I know what is going on here, the `DOUBLE_EPSILON` is `1.11e-16`, however, in this case we need a less precise epsilon, otherwise the logic of rounding would be broken and we'll get (or won't get) extra +1 column on the east. Look also into this comment: https://github.com/locationtech/geotrellis/blob/master/raster/src/main/scala/geotrellis/raster/GridExtent.scala#L382-L388

GDAL uses 1e-5, QGIS 1e-6 and we're using 1e-7. It looks like it is a heuristic in all libs.

Reply:
@pomadchin is right!  The epsilons in `Constants` are  good baseline to use when you don't know enough about the problem you're working on.  The predefined values are the sort of best practice values for numerical methods (think computing square roots via Newton's method or something similar).  When thresholding for edge cases, other fudge factors may be appropriate, which we set using case-by-case heuristics (_cough_ _cough_ guessing _cough_).  In those cases, it may be appropriate to comment above the definition of the buffer value that it is an intentional choice and why.